src/ocean/meta/values/Reset_test.d

```diff
@@ -0,0 +1,183 @@
+                    this.j.length = 3;
+                    this.j[0].copy("abc");
+                    this.j[1].copy("def");
+                    this.j[2].copy("ghi");
+                    foreach ( ref item; this.k )
+                    {
+                        item = 120000;
+                    }
+                }
+            }
+
+            void InitStructure()
+            {
+                this.d = 32;
+                this.e.copy("even more test text");
+
+                this.f.length = 1;
+                this.f[0].copy("abc");
+                foreach ( ref item; this.g )
+                {
+                    item = 32400;
+                }
+            }
+
+            SubSubStruct[] sub_sub_struct;
+        }
+
+        SubStruct sub_struct;
+
+        SubStruct[] sub_struct_array;
+    }
+
+    TestStruct test_struct;
+    test_struct.a = 7;
+    test_struct.b.copy("some test");
+    foreach ( i, ref item; test_struct.c )
+    {
+        item = 64800;
+    }
+
+    TestStruct.SubStruct sub_struct;
+    sub_struct.InitStructure;
+    test_struct.sub_struct = sub_struct;
+    test_struct.sub_struct_array ~= sub_struct;
+    test_struct.sub_struct_array ~= sub_struct;
+
+
+    TestStruct.SubStruct.SubSubStruct sub_sub_struct;
+    sub_sub_struct.InitStructure;
+    test_struct.sub_struct_array[0].sub_sub_struct ~= sub_sub_struct;
+    test_struct.sub_struct_array[1].sub_sub_struct ~= sub_sub_struct;
+    test_struct.sub_struct_array[1].sub_sub_struct ~= sub_sub_struct;
+    test_struct.sub_struct.sub_sub_struct ~= sub_sub_struct;
+    test_struct.sub_struct.sub_sub_struct ~= sub_sub_struct;
+
+    reset(test_struct);
+
+    test!("==")(test_struct.a, 0);
+    test!("==")(test_struct.b, ""[]);
+    foreach ( item; test_struct.c )
+    {
+        test!("==")(item, 0);
+    }
+
+    test!("==")(test_struct.sub_struct_array.length, 0);
+
+    test!("==")(test_struct.sub_struct.d, 0);
+    test!("==")(test_struct.sub_struct.e, ""[]);
+    test!("==")(test_struct.sub_struct.f.length, 0);
+    foreach ( item; test_struct.sub_struct.g )
+    {
+        test!("==")(item, 0);
+    }
+
+    test!("==")(test_struct.sub_struct.sub_sub_struct.length, 0);
+
+    //Test nested classes.
+    class TestClass
+    {
+        int a;
+        char[] b;
+        int[2] c;
+
+        public class SubClass
+        {
+            int d;
+            char[] e;
+        }
+
+        SubClass s;
+    }
+
+    TestClass test_class = new TestClass;
+    test_class.s =  test_class.new SubClass;
+    test_class.a = 7;
+    test_class.b = [];
+    test_class.b ~= 't';
+    test_class.c[1] = 1;
+    test_class.s.d = 5;
+    test_class.s.e = [];
+    test_class.s.e ~= 'q';
+
+    reset(test_class);
+    test!("==")(test_class.a, 0);
+    test!("==")(test_class.b.length, 0);
+    test!("==")(test_class.c[1], 0);
+    test!("!is")(cast(void*)test_class.s, null);
+    test!("==")(test_class.s.d, 0);
+    test!("==")(test_class.s.e.length, 0);
+}
+
+unittest
```

Comment:
What does this test check that the other one doesn't? A comment would be helpful.

Reply:
I removed it - it was simply copied from a different place but it doesn't check anything extra.